waspc/src/Analyzer/StdTypeDefinitions.hs

```diff
@@ -12,17 +13,21 @@ module Analyzer.StdTypeDefinitions
   )
 where
 
+import Analyzer.Evaluator.TH (makeDecl, makeEnum)
 import qualified Analyzer.TypeDefinitions as TD
-import GHC.Generics (Generic)
 
-data AuthMethod = EmailAndPassword deriving (Generic)
+data AuthMethod = EmailAndPassword deriving (Show, Eq)
 
-data App = App {title :: String, authMethod :: AuthMethod} deriving (Generic)
+data App = App {title :: String, authMethod :: AuthMethod} deriving (Show, Eq)
```

Comment:
I was thinking we might get circular dependencies in that case, but since you split the TypeDefinitions into multiple modules that probably won't be a problem! Ok cool.
Can you please update the TODO with information I mentioned above?

Reply:
Added TODO.